examples/nidcpower_source_dc_voltage/measurement.py

```diff
@@ -68,96 +66,91 @@ def measure(
     """Source and measure a DC voltage with an NI SMU."""
     logging.info("Executing measurement: pin_names=%s voltage_level=%g", pin_names, voltage_level)
 
-    session_management_client = nims.session_management.Client(
-        grpc_channel=measurement_service.get_channel(
-            provided_interface=nims.session_management.GRPC_SERVICE_INTERFACE_NAME,
-            service_class=nims.session_management.GRPC_SERVICE_CLASS,
-        )
+    session_management_client = create_session_management_client(measurement_service)
+
+    reservation = reserve_session(
+        session_management_client,
+        measurement_service.context.pin_map_context,
+        timeout=60,
+        pin_names=pin_names,
     )
 
-    with contextlib.ExitStack() as stack:
-        reservation = stack.enter_context(
-            session_management_client.reserve_sessions(
-                context=measurement_service.context.pin_map_context,
-                pin_or_relay_names=pin_names,
-                instrument_type_id=nims.session_management.INSTRUMENT_TYPE_NI_DCPOWER,
-                # If another measurement is using the session, wait for it to complete.
-                # Specify a timeout to aid in debugging missed unreserve calls.
-                # Long measurements may require a longer timeout.
-                timeout=60,
-            )
+    if len(reservation.session_info) != 1:
+        measurement_service.context.abort(
+            grpc.StatusCode.INVALID_ARGUMENT,
+            f"Unsupported number of sessions: {len(reservation.session_info)}",
         )
 
-        if len(reservation.session_info) != 1:
-            measurement_service.context.abort(
-                grpc.StatusCode.INVALID_ARGUMENT,
-                f"Unsupported number of sessions: {len(reservation.session_info)}",
-            )
+    session_grpc_channel = get_grpc_device_channel(measurement_service, nidcpower)
+    
+    session = create_nidcpower_session(
```

Comment:
This leaks the session, which may leave driver resources reserved. Use `with create_session(..) as session` to manage the lifetime of the session.

Reply:
Enclosed session creation code inside `with`